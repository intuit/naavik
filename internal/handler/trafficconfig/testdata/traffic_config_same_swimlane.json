{
    "apiVersion": "admiral.io/v1",
    "kind": "TrafficConfig",
    "metadata": {
      "annotations": {
        "revisionNumber": "89",
        "transactionID": "1-65787080-66081ef70d07abc4697045ac"
      },
      "labels": {
        "asset": "httpbin",
        "env": "e2e"
      },
      "name": "httpbin-e2e",
      "namespace": "admiral"
    },
    "spec": {
      "edgeService": {
        "filters": [
          {
            "name": "filter-3387192"
          },
          {
            "name": "filter--671448791"
          }
        ],
        "routes": [
          {
            "config": [
              {
                "targetGroupSelector": "Default",
                "targetSelector": "swc"
              }
            ],
            "filterSelector": "filter--671448791",
            "inbound": "/v1/*",
            "name": "v1",
            "outbound": "/*",
            "timeout": 5000,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-air"
            ]
          },
          {
            "config": [
              {
                "targetGroupSelector": "sw1",
                "targetSelector": "swb"
              },
              {
                "targetGroupSelector": "sw2",
                "targetSelector": "swb"
              },
              {
                "targetGroupSelector": "sw3",
                "targetSelector": "swb"
              }
            ],
            "filterSelector": "filter--671448791",
            "inbound": "/*",
            "name": "all",
            "outbound": "/*",
            "timeout": 5000,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-air"
            ]
          },
          {
            "config": [
              {
                "targetGroupSelector": "sw1",
                "targetSelector": "swb"
              },
              {
                "targetGroupSelector": "sw2",
                "targetSelector": "swb"
              },
              {
                "targetGroupSelector": "sw3",
                "targetSelector": "swb"
              }
            ],
            "filterSelector": "filter-3387192",
            "inbound": "/health/full",
            "name": "Health Check",
            "outbound": "/health/fullrewrite",
            "timeout": 5000,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-air"
            ]
          }
        ],
        "targetGroups": [
          {
            "name": "Default - Single Swimlane",
            "weights": [
              {
                "name": "Default - All Traffic",
                "weight": 100
              }
            ]
          },
          {
            "appOverrides": [
              {
                "assetAlias": "httpbin",
                "assetID": "2093065520108338843",
                "weights": [
                  {
                    "name": "swc",
                    "weight": 100
                  },
                  {
                    "name": "swb",
                    "weight": 0
                  }
                ]
              },
              {
                "assetAlias": "httpbin",
                "assetID": "5047426875663384589",
                "weights": [
                  {
                    "name": "swb",
                    "weight": 100
                  },
                  {
                    "name": "swc",
                    "weight": 0
                  }
                ]
              }
            ],
            "name": "Swb-swc",
            "weights": [
              {
                "name": "swb",
                "weight": 100
              },
              {
                "name": "swc",
                "weight": 0
              }
            ]
          },
          {
            "name": "DefaultGroup",
            "weights": [
              {
                "name": "Default",
                "weight": 100
              }
            ]
          },
          {
            "name": "3 sws",
            "weights": [
              {
                "name": "sw2",
                "weight": 50
              },
              {
                "name": "sw3",
                "weight": 30
              },
              {
                "name": "sw1",
                "weight": 20
              }
            ]
          }
        ],
        "targets": [
          {
            "meshDNS": "e2e-air.httpbin.mesh",
            "name": "e2e-540681375",
            "port": 80,
            "socketTimeout": 5000
          },
          {
            "meshDNS": "e2e-air.httpbin.mesh",
            "name": "swc",
            "port": 80,
            "socketTimeout": 5000
          },
          {
            "meshDNS": "qal.httpbineast.mesh",
            "name": "swe",
            "port": 80,
            "socketTimeout": 5000
          },
          {
            "meshDNS": "e2e-air.httpbin.mesh",
            "name": "swb",
            "port": 80,
            "socketTimeout": 5000
          },
          {
            "meshDNS": "qal.httpbin.mesh",
            "name": "swd",
            "port": 80,
            "socketTimeout": 5000
          }
        ]
      },
      "quotaGroup": {
        "totalQuotaGroups": [
          {
            "cpuLimit": 80,
            "description": "Total capacity of service",
            "memoryLimit": 80,
            "name": "Total Throttling Plan",
            "podLevelThreshold": 100,
            "quotas": [
              {
                "algorithm": "scalable",
                "behaviour": "softThrottle",
                "keyType": "any",
                "maxAmount": 2001,
                "name": "Total",
                "rule": "/*",
                "timePeriod": "1s"
              }
            ],
            "regionLevelLimit": true,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-air"
            ]
          }
        ]
      },
      "workloadEnvs": [
        "e2e",
        "e2e-air"
      ]
    }
  }