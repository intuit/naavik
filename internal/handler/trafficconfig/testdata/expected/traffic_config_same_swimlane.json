{
  "hosts": [
    "e2e.httpbin.mesh",
    "e2e.httpbin.intuit",
    "e2e-air.httpbin.mesh"
  ],
  "http": [
    {
      "name": "v1",
      "match": [{ "headers": { "x-intuit-route-name": { "exact": "v1" } } }],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 100
        }
      ],
      "timeout": "10s"
    },
    {
      "name": "all",
      "match": [{ "headers": { "x-intuit-route-name": { "exact": "all" } } }],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 20
        },
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 50
        },
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 30
        }
      ],
      "timeout": "10s"
    },
    {
      "name": "Health Check",
      "match": [
        { "headers": { "x-intuit-route-name": { "exact": "Health Check" } } }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 20
        },
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 50
        },
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          },
          "weight": 30
        }
      ],
      "timeout": "10s"
    },
    {
      "name": "defaultall-e2e",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "e2e.httpbin.mesh" }
        }
      ],
      "route": [
        { "destination": { "host": "e2e.httpbin.mesh" } }
      ],
      "timeout": "10s"
    },
    {
      "name": "defaultall-e2e-air-mesh",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "e2e-air.httpbin.mesh" }
        }
      ],
      "route": [
        { "destination": { "host": "e2e-air.httpbin.mesh" } }
      ],
      "timeout": "10s"
    },
    {
      "name": "defaultall-e2e-air",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "e2e.httpbin.intuit" }
        }
      ],
      "route": [
        { "destination": { "host": "e2e-air.httpbin.mesh" } }
      ],
      "timeout": "10s"
    }
  ]
}
